docs/framework/guides/architecture/ui-components.md

```diff
@@ -0,0 +1,89 @@
+---
+category: framework-architecture
+order: 46
+---
+
+# UI components
+
+All UI classes come with `set()` method, which sets the properties of the components, such as labels, icons, placeholders, etc.
+
+### Buttons
+
+There are two basic buttons in CKEditor 5 UI library - a button and a switch button.
+
+{@snippet framework/ui/ui-buttons}
+
+Button is implemented with the {@link module:ui/button/buttonview~ButtonView `ButtonView`} class.
+
+```js
```

Comment:
Let's make one block of code per one example as it is done usually in UI libraries